docs/developing.rst

```diff
@@ -0,0 +1,199 @@
+====================
+Developer Guidelines
+====================
+
+This webpage is intended to guide users through making making changes to
+``tedana``'s codebase, in particular working with tests.
+The worked example also offers some guidelines on approaching testing when
+adding new functions.
+
+Adding and Modifying Tests
+==========================
+Testing is an important component of development.
+For simplicity, we have migrated all tests to ``pytest``.
+There are two basic kinds of tests: unit and integration tests.
+Unit tests focus on testing individual functions, whereas integration tests focus on making sure
+that the whole workflow runs correctly.
+
+For unit tests, we try to keep tests on the same module grouped into one file.
+Make sure the function you're testing is imported, then write your test.
+Good tests will make sure that edge cases are accounted for as well as common cases.
+You may also use ``pytest.raises`` to ensure that errors are thrown for invalid inputs to a
+function.
+
+For integration tests, make a ``tar.gz`` file which will unzip to be only the files you'd like to
```

Comment:
I would couch this -- it's going to be fairly uncommon at this point. I might say something like:

> In rare cases, you may need to add a new integration test. This means adding a new kind of dataset that tedana will be explicitly tested to support. As always, please open an issue first if you think that you have such a dataset ! Then, after making sure you have appropriate permission to share it, follow these steps to add it to the tedana repository.

Reply:
Also, this should be a new section !

Reply:
Addressed in 6a6a71f and 13e3daa